src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -109,11 +113,24 @@ public function isStrictRequirements()
      */
     public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
     {
-        if (null === $route = $this->routes->get($name)) {
+        $locale = $parameters['_locale']
+            ?? $this->context->getParameter('_locale')
+            ?: $this->defaultLocale;
```

Comment:
Based on what I see in the constructor, this->defaultLocale could be null, no?

Reply:
Correct, which results in a route not found.

Reply:
it still produces an implicit route lookup, i.e. `"route."`.

Instead why not normalize `$name` and keep the route lookup below as is...

```php
if (null !== $locale) {
   $name .= '.'.$locale;
}
```

that would do no?